### PR TITLE
Rename the Period class to Interval

### PR DESCRIPTION
**Pull request by sdispater**
## Pull Request Check List

This change is both semantic change and breaking change.

This renames the `Period` class to `Interval` to make its purpose and what it represents clearer.

- [x] Added **tests** for changed code.
- [ ] Updated **documentation** for changed code.

### CODE REVIEW

File: pendulum/date.py [hunk 1]
```diff
@@ -283,7 +283,7 @@ def __sub__(self, other: timedelta | date) -> Date | Period:
 
     # DIFFERENCES
 
-    def diff(self, dt: date | None = None, abs: bool = True) -> Period:
+    def diff(self, dt: date | None = None, abs: bool = True) -> Interval:
         """
         Returns the difference between two Date objects as a Period.
```
Comment (Secrus) [inline pendulum/date.py:288]:
```suggestion
        Returns the difference between two Date objects as an Interval.
```

File: pendulum/__init__.py [hunk 2]
```diff
@@ -286,11 +286,11 @@ def duration(
     )
 
 
-def period(start: DateTime, end: DateTime, absolute: bool = False) -> Period:
+def interval(start: DateTime, end: DateTime, absolute: bool = False) -> Interval:
     """
     Create a Period instance.
```
Comment (Secrus) [inline pendulum/__init__.py:291]:
```suggestion
    Create an Interval instance.
```

**Review by Secrus — CHANGES_REQUESTED** [review]

File: pendulum/date.py (hunk 1 above)
Comment (sdispater) [reply pendulum/date.py]: Done!

File: pendulum/__init__.py (hunk 2 above)
Comment (sdispater) [reply pendulum/__init__.py]: Done!

**Review by Secrus — APPROVED** [review]

**Pull request merged**